components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/internal/StreamProcessorDeployer.java

```diff
@@ -145,6 +156,124 @@ private static String getFileNameWithoutExtenson(String fileName) {
         return fileName;
     }
 
+    private static SiddhiAppType getArtifactType(String fileName) {
+        fileName = fileName.toUpperCase();
+        if (fileName.startsWith(SiddhiAppProcessorConstants.APIM_SIDDHI_APP_PREFIX)) {
+            return SiddhiAppType.APIM;
+        } else if (fileName.startsWith(SiddhiAppProcessorConstants.EI_SIDDHI_APP_PREFIX)) {
+            return SiddhiAppType.EI;
+        } else if (fileName.startsWith(SiddhiAppProcessorConstants.IS_SIDDHI_APP_PREFIX)) {
+            return SiddhiAppType.IS;
+        } else {
+            return SiddhiAppType.OTHER;
+        }
+    }
+
+    private static boolean isDeploymentAllowed(SiddhiAppType siddhiAppType) {
+        switch (serverType) {
+            case SP:
+                switch (siddhiAppType) {
+                    case OTHER:
+                        return true;
+                    case APIM:
+                        if (apimAnalyticsEnabledOnSP) {
+                            return true;
+                        }
+                        break;
+                    case IS:
+                        if (isAnalyticsEnabledOnSP) {
+                            return true;
+                        }
+                        break;
+                    case EI:
+                        if (eiAnalyticsEnabledOnSP) {
+                            return true;
+                        }
+                        break;
+                }
+                break;
+            case APIM:
+                if (siddhiAppType.name().equals(SiddhiAppType.APIM.name())) {
+                    return true;
+                }
+                break;
+            case IS:
+                if (siddhiAppType.name().equals(SiddhiAppType.IS.name())) {
+                    return true;
+                }
+                break;
+            case EI:
+                if (siddhiAppType.name().equals(SiddhiAppType.EI.name())) {
+                    return true;
+                }
+                break;
+        }
+        return false;
+    }
+
+    private void setServerType() {
+        ConfigProvider configProvider = StreamProcessorDataHolder.getInstance().getConfigProvider();
+        if (configProvider != null) {
+            try {
+                LinkedHashMap analyticsSolutionsMap = (LinkedHashMap) configProvider.
+                        getConfigurationObject(SiddhiAppProcessorConstants.ANALYTICS_SOLUTIONS);
+                if (analyticsSolutionsMap != null) {
+                    Object typeObject = analyticsSolutionsMap.get(
```

Comment:
should be top level element add next to the server-id or name field

Reply:
Fixed in  e3e2a00